### PR TITLE
Fix table sorting after modal submit

### PR DESCRIPTION
**Pull request by tpapaioa**
# Description

After submitting a modal (add to group, remove from group, or update) from the Systems page, the refreshed table is sorted by name instead of by "Last seen", which is the default. Also, the "Last seen" column header is still highlighted, as if the table is still sorted by that column.

Similar behavior can be seen on the Groups, Repositories, and Images pages too.

This PR fixes this issue by setting `hasModalSubmitted` when modals have submitted, and re-running the relevant query w/ the correct sort and pagination query params in `GeneralTable` when this state variable has changed.

Fixes # THEEDGE-2541

## Type of change

What is it?

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update

# Checklist:

- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I run `npm run lint:js:fix` to check that my code is properly formatted

### CODE REVIEW

**Comment by matoval** [comment]
I think we can resolve this issue with default queries in the api files. We are doing this for the image table api calls:  https://github.com/RedHatInsights/edge-frontend/blob/master/src/api/images/index.js#L73

**Comment by tpapaioa** [comment]
> I think we can resolve this issue with default queries in the api files. We are doing this for the image table api calls: https://github.com/RedHatInsights/edge-frontend/blob/master/src/api/images/index.js#L73

That would work for the default parameters, but not if the user has changed the sort/pagination/filter settings. After submitting the modal, those settings are still displayed, but the displayed table data don't respect those settings prior to this fix.

File: src/Routes/Devices/Inventory.js [hunk 1]
```diff
@@ -50,6 +50,11 @@ const Inventory = () => {
     setIsRowSelected(isRow);
   };
 
+  const reloadData = () => {
+    fetchDevices();
+    setTimeout(() => setHasModalSubmitted(true), 800);
```
Comment (matoval) [inline src/Routes/Devices/Inventory.js:55]: We should replace all setTimeout with async/await this way we don't have to set a hard coded wait time.

File: src/components/general-table/GeneralTable.js [hunk 2]
```diff
@@ -120,7 +120,7 @@ const GeneralTable = ({
       : apiFilterSort
       ? loadTableData(dispatch, query)
       : null;
-  }, [chipsArray, perPage, page, sortBy]);
+  }, [chipsArray, perPage, page, sortBy, hasModalSubmitted]);
```
Comment (matoval) [inline src/components/general-table/GeneralTable.js:123]: Adding hasModalSubmitted is causing a double render, once when it's set to true and another when it's set back to false.

**Review by matoval — CHANGES_REQUESTED** [review]

**Review by resoluteCoder — APPROVED** [review]
We can merge but we can create a bug ticket to remove the double render. At least this will fix a user impacting bug. Good job @tpapaioa

**Review by matoval — APPROVED** [review]

**Pull request merged**